server/game/cards/21-FtR/BattleoftheTrident.js

```diff
@@ -0,0 +1,111 @@
+const AgendaCard = require('../../agendacard');
+const RevealPlots = require('../../gamesteps/revealplots');
+const SimpleStep = require('../../gamesteps/simplestep.js');
+
+class BattleoftheTrident extends AgendaCard {
+    setupCardAbilities(ability) {                
+        this.reaction({
+            when: {
+                onClaimApplied: event => 
+                    event.challenge.winner === this.owner &&
+                    event.challenge.attackingPlayer === this.owner &&
+                    event.challenge.attackers.some(attacker => attacker.hasTrait('Army') || attacker.hasTrait('Commander')) &&
+                    !this.owner.hasFlag('cannotRevealPlot')
+            },
+            cost: ability.costs.kneelFactionCard(),
+            target: {
+                type: 'select',
+                activePromptTitle: 'Select a plot',
+                cardCondition: (card, context) => 
+                    card.controller === context.player && 
+                    card.location === 'plot deck' && 
+                    card.hasTrait(this.translateChallengeTypeToTrait(context.event.challenge.challengeType)),
+                cardType: 'plot'
+            },
+            handler: context => this.trigger(context)
+        });
+
+        this.action({
+            title: 'Manually trigger',
+            cost: ability.costs.kneelFactionCard(),
+            condition: () => !this.owner.hasFlag('cannotRevealPlot'),
+            target: {
+                type: 'select',
+                activePromptTitle: 'Select a plot',
+                //do not filter the plotTrait when triggering manually
+                cardCondition: (card, context) => 
+                    card.controller === context.player && 
+                    card.location === 'plot deck',
+                cardType: 'plot'
+            },
+            handler: context => this.trigger(context)
+        });
+
+        //this reaction is not intended to trigger, only to be used directly by the trigger() function
+        this.reaction({
+            title: 'removeFromGameMarker',
+            when: {
+                onClaimApplied: () => false
+            },
```

Comment:
I do think this is fairly weird, though clever. I think the "correct" thing would be to convert revealing a plot into a game action, and have the rest of it as a `then` action. But I suspect that would be a major engine undertaking, so I think we'll just merge this as-is.

Reply:
good idea about converting it into a gameaction. i´d rather not do that work now and i´ve added a TODO on this part and will create an issue for it.